cypress/integration/Dialog.spec.ts

```diff
@@ -1,11 +1,198 @@
 describe('Dialog', () => {
   beforeEach(() => {
-    cy.visitStory('dialog--styled');
+    cy.visitStory('dialog--cypress');
   });
 
-  it('should open and close correctly', () => {
-    cy.findByText('open').click();
-    cy.findByText('close').click();
-    cy.findByText('close').should('not.exist');
+  function shouldBeOpen() {
+    cy.findByText('title').should('exist');
+  }
+  function shouldBeClosed() {
+    cy.findByText('title').should('not.exist');
+  }
+  function shouldNotAllowOutsideInteraction(action: 'realClick' | 'realTouch') {
+    cy.findByLabelText('count up')
+      .invoke('text')
+      .then((count) => {
+        if (action === 'realTouch') {
+          cy.findByLabelText('count up').realClick();
+        } else {
+          cy.findByLabelText('count up').realTouch();
+        }
+        cy.findByLabelText('count up').should('have.text', count);
+      });
+  }
+  function shouldAllowOutsideInteraction(action: 'realClick' | 'realTouch') {
+    cy.findByLabelText('count up')
+      .invoke('text')
+      .then((count) => {
+        if (action === 'realTouch') {
+          cy.findByLabelText('count up').realClick();
+        } else {
+          cy.findByLabelText('count up').realTouch();
+        }
+        cy.findByLabelText('count up').should('not.eq', count);
```

Comment:
Is this asserting what we expect? `count` is `text` but is `cy.findByLabelText('count up')` a button? will they always **not** be equal?

Reply:
hmm, good shout, let me check on that.

Reply:
You were 💯 on the money here, great catch! I made it fail and it was still passing. Fixed now: https://github.com/radix-ui/primitives/pull/1401/commits/4d941df0a6aa728c0f031cd7df76883465861966